frame/vamm/src/lib.rs

```diff
@@ -62,6 +62,45 @@
 //!
 //! <!-- Original author: @Cardosaum -->
 
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+// Allow some linters for tests.
+#![cfg_attr(
+	not(test),
+	warn(
+		clippy::disallowed_methods,
+		clippy::disallowed_types,
+		clippy::indexing_slicing,
+		clippy::todo,
+		clippy::unwrap_used,
+		clippy::panic,
+		clippy::doc_markdown
+	)
+)]
+// Specify linters to VAMM Pallet.
+#![warn(clippy::unseparated_literal_suffix)]
+#![deny(
```

Comment:
Curious to know how you chose these lints

Reply:
It was a mix of experimentation using `clippy::pedantic` to discover new interesting linters and I also took a look in some linters being used across Composable pallets.